core/controller/src/main/scala/whisk/core/controller/actions/PrimitiveActions.scala

```diff
@@ -258,7 +257,7 @@ protected[actions] object ActivationFinisher {
     // when the future completes, self-destruct
     promise.future.andThen { case _ => shutdown() }
 
-    val preemptiveMsgs: Buffer[Cancellable] = Buffer.empty
+    var preemptiveMsgs = Vector.empty[Cancellable]
```

Comment:
Mega nit: Should this be an `immutable.Buffer` for its constant append time?

Reply:
There isn't an Immutable Buffer in the Scala collection library. 

[`Vector` is effectively constant on all operations](http://docs.scala-lang.org/overviews/collections/performance-characteristics.html) and the best alternative in the immutable side.

Reply:
Gnah, I of course meant the immutable.Queue. My bad.

Reply:
You're right; we are only appending to the datastructure. `Queue`'s append is constant time versus `Vector`s which is effectively constant.

Reply:
As you say, performance issue isn't there so I'd probably still go with a `Vector` personally.